torch/csrc/api/src/nn/modules/linear.cpp

```diff
@@ -30,28 +30,33 @@ LinearImpl::LinearImpl(const LinearOptions& options_) : options(options_) {
 }
 
 void LinearImpl::reset() {
-  weight =
-      register_parameter("weight", torch::empty({options.out(), options.in()}));
-  if (options.with_bias()) {
-    bias = register_parameter("bias", torch::empty(options.out()));
+  weight = register_parameter("weight", 
+    torch::empty({options.out_features(), options.in_features()}));
+  if (options.bias()) {
+    bias = register_parameter("bias", torch::empty(options.out_features()));
+  } else {
+    bias = register_parameter("bias", {}, /*requires_grad=*/false);
   }
 
-  const auto stdv = 1.0 / std::sqrt(weight.size(1));
-  NoGradGuard no_grad;
-  for (auto& p : this->parameters()) {
-    p.uniform_(-stdv, stdv);
+  torch::nn::init::kaiming_uniform_(weight, sqrt(5));
```

Comment:
clang-tidy is acting up again :(
```
{
    path: 'torch/csrc/api/src/nn/modules/linear.cpp',
    start_line: 41,
    end_line: 41,
    start_column: 50,
    end_column: 50,
    annotation_level: 'failure',
    message: '[[cppcoreguidelines-avoid-magic-numbers]] warning: 5 is a ' +
      'magic number; consider replacing it with a named constant'
  }
```
Let's probably add `// NOLINT(cppcoreguidelines-avoid-magic-numbers)` at the end of this line :D